### PR TITLE
Lucien/topic list checkbox

### PR DESCRIPTION
**Pull request by edellucien**
- Add Topic List which generates a layer tree per topic
- Add generic Checkbox/radio component to style TopicList and LayerTree

### CODE REVIEW

File: src/ConfigReader.js [hunk 1]
```diff
@@ -116,16 +116,18 @@ class ConfigReader {
     layer.init(this.map);
 
     if (item.children) {
-      item.children.forEach(childConf => {
-        layer.addChild(this.loadLayerFromConfig(childConf));
+      Object.keys(item.children).forEach(childConf => {
```
Comment (oterral) [inline src/ConfigReader.js]: Bad merge we want to keep the array for children.

File: src/ConfigReader.js [hunk 2]
```diff
@@ -116,16 +116,18 @@ class ConfigReader {
...
+        layer.addChild(this.loadLayerFromConfig(item.children[childConf]));
       });
     }
 
     return layer;
   }
 
   initialize() {
-    return this.data.map(val => this.loadLayerFromConfig(val));
+    return Object.keys(this.data).map(val =>
```
Comment (oterral) [inline src/ConfigReader.js]: here too

File: src/components/Checkbox/Checkbox.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  /**
+   * HTML tabIndex attribute
+   */
+  tabIndex: PropTypes.number,
+
+  /**
+   * Key of the label
+   */
+  keyName: PropTypes.string,
```
Comment (oterral) [inline src/components/Checkbox/Checkbox.js]: ambigous name , do you mean a `key` attribute for react ?

Comment (oterral) [reply src/components/Checkbox/Checkbox.js]: or an html  ` name` attribute

File: src/components/Checkbox/Checkbox.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      className,
+      onClick,
+      tabIndex,
+      inputType,
+      checked,
+      keyName,
+    } = this.props;
+
+    return (
+      <label // eslint-disable-line
```
Comment (oterral) [inline src/components/Checkbox/Checkbox.js]: Why disabling eslint ?

File: src/components/Checkbox/Checkbox.scss [hunk 5]
```diff
@@ -0,0 +1,69 @@
+label {
```
Comment (oterral) [inline src/components/Checkbox/Checkbox.scss]: Can you encapsulate all styles in `tm-checkbox` pls?

File: src/components/Checkbox/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+#
+
+This demonstrates the use of Checkbox.
+
+```jsx
+const React = require('react');
+const Checkbox = require('./Checkbox').default;
```
Comment (oterral) [inline src/components/Checkbox/README.md]: You can use  es6 import now witht he last version a react-styleguidist  `import React from 'react' ` and  `import Checkbox from 'react-spatial/components/Checkbox' `

File: src/components/LayerTree/LayerTree.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import Checkbox from '../Checkbox/Checkbox';
```
Comment (oterral) [inline src/components/LayerTree/LayerTree.js]: Only `../Checkbox`  is enough

File: src/components/LayerTree/LayerTree.md.scss [hunk 8]
```diff
@@ -12,9 +18,83 @@
     margin: auto;
   }
 
-  .tm-map {
-    width: 100%;
-    height: 100%;
-    flex-grow: 1;
+  .tm-layer-tree-arrow {
```
Comment (oterral) [inline src/components/LayerTree/LayerTree.md.scss]: Where does come from all these style ? They weren't there before?

File: src/components/TopicList/TopicList.js [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+      visible: PropTypes.bool,
+      expanded: PropTypes.bool,
+      children: PropTypes.array,
+    }),
+  ).isRequired,
+
+  /**
+   * All props aimed to be passed to child LayerTree
+   */
+  propsToLayerTree: PropTypes.object.isRequired,
```
Comment (oterral) [inline src/components/TopicList/TopicList.js]: Not really required we could use default values.no?

File: src/components/Checkbox/Checkbox.js (hunk 4 above)
Comment (edellucien) [reply src/components/Checkbox/Checkbox.js]: I can fix this with 'htmlFor' on label

File: src/components/LayerTree/LayerTree.js (hunk 7 above)
Comment (edellucien) [reply src/components/LayerTree/LayerTree.js]: I see magic happening everywhere ;)

File: src/components/LayerTree/LayerTree.md.scss (hunk 8 above)
Comment (edellucien) [reply src/components/LayerTree/LayerTree.md.scss]: duplicated due to merge

**Comment by edellucien** [comment]
Added index.js for Topiclist and render topiclist barrierfree

**Pull request merged**